godot-core/src/builtin/file.rs

```diff
@@ -0,0 +1,459 @@
+            FileAccess::open_encrypted_with_pass(path.clone(), flags, pass).ok_or_else(|| {
+                IoError::new(
+                    ErrorKind::Other,
+                    format!(
+                        "Can't open file {} in mode {:?}; GodotError: {:#?}",
+                        &path,
+                        flags,
+                        FileAccess::get_open_error()
+                    ),
+                )
+            })?;
+
+        Ok(Self::init(fa))
+    }
+
+    /// Closes the inner [Gd]<[FileAccess]> and destroys the [GFile] instance
+    pub fn close(mut self) {
+        self.fa.close()
+    }
+
+    /// Deconstructs the [GFile] intance and retrieves the inner [Gd]<[FileAccess]>
+    pub fn into_inner(self) -> Gd<FileAccess> {
+        self.fa
+    }
+
+    // Remaps of the internal FileAccess methods
+
+    // Reads
+
+    /// Reads the next 8 bits from the file as [u8]. Exposed Godot method.
+    pub fn get_8(&self) -> IoResult<u8> {
+        let val = self.fa.get_8();
+        self.check_error()?;
+        Ok(val)
+    }
+
+    /// Reads the next 16 bits from the file as [u16].
+    /// For details check the [Godot documentation](https://docs.godotengine.org/en/stable/classes/class_fileaccess.html)
+    pub fn get_16(&self) -> IoResult<u16> {
+        let val = self.fa.get_16();
+        self.check_error()?;
+        Ok(val)
+    }
+
+    /// Reads the next 32 bits from the file as [u32]. Exposed Godot method.
+    /// For details check the [Godot documentation](https://docs.godotengine.org/en/stable/classes/class_fileaccess.html)
+    pub fn get_32(&self) -> IoResult<u32> {
+        let val = self.fa.get_32();
+        self.check_error()?;
+        Ok(val)
+    }
+
+    /// Reads the next 64 bits from the file as [u64]. Exposed Godot method.
+    /// For details check the [Godot documentation](https://docs.godotengine.org/en/stable/classes/class_fileaccess.html)
+    pub fn get_64(&self) -> IoResult<u64> {
+        let val = self.fa.get_64();
+        self.check_error()?;
+        Ok(val)
+    }
+
+    /// Reads the next 64 bits from the file as [u64]. Exposed Godot method.
+    /// For details check the [Godot documentation](https://docs.godotengine.org/en/stable/classes/class_fileaccess.html)
+    pub fn get_pascal_string(&mut self) -> IoResult<GodotString> {
+        let val = self.fa.get_pascal_string();
+        self.check_error()?;
+        Ok(val)
+    }
+
+    /// Reads the whole file as [GodotString]. Text is interpeted as `UTF-8` encoded.
+    /// If `skip_cr` is set to `true`, carriage returns will be ingored when parsing
+    /// and only line feed characters will represents a new line
+    /// For details check the [Godot documentation](https://docs.godotengine.org/en/stable/classes/class_fileaccess.html)
+    pub fn get_as_text(&self, skip_cr: bool) -> IoResult<GodotString> {
+        let val = self.fa.get_as_text_ex().skip_cr(skip_cr).done();
+        self.check_error()?;
+        Ok(val)
+    }
```

Comment:
I would name this `read_text_entire` and maybe also link to this Rust function:
https://doc.rust-lang.org/std/io/fn.read_to_string.html